git/github/github.go

```diff
@@ -302,6 +317,83 @@ func (c *Client) GetCheckRunID(ctx context.Context, owner string, repo string, S
 	return nil, nil
 }
 
+// GetAllCheckRunsForRef returns all existing GitHub CheckRuns if a match for the Owner, Repo, SHA, and appID.
+func (c *Client) GetAllCheckRunsForRef(ctx context.Context, owner string, repo string, SHA string, appID int64) ([]*ghapi.CheckRun, error) {
+	filter := "all"
+
+	res, _, err := c.GetChecksService().ListCheckRunsForRef(
+		ctx,
+		owner,
+		repo,
+		SHA,
+		&ghapi.ListCheckRunsOptions{
+			AppID:  &appID,
+			Filter: &filter,
+		},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if *res.Total == 0 {
+		c.logger.Info("Found no CheckRuns for the ref", "SHA", SHA)
+		return nil, nil
+	}
+
+	return res.CheckRuns, nil
+}
+
+// GetCheckRunIDAndIfUpdateNeeded returns an existing GitHub CheckRun ID if a match is found and if updated/create is needed for the SHA, externalID and appID.
+func (c *Client) GetCheckRunIDAndIfUpdateNeeded(checkRuns []*ghapi.CheckRun, checkRun *CheckRunAdapter) (*int64, bool, error) {
+	for _, cr := range checkRuns {
+		cr := cr
+		if *cr.ExternalID == checkRun.ExternalID {
+
+			if checkRun.CompletionTime.After((*cr.CompletedAt).Time) {
+				// We need to update the existing checkrun if the completionTime is after the gotten checkrun's CompletionTime
+				c.logger.Info("Found CheckRun with a matching ExternalID and completedAt not later than the CompletedAt ", "CompletedAt", checkRun.CompletionTime)
+				return cr.ID, true, nil
+			} else {
+				// We don't need to update the existing checkrun if the completionTime is not after the gotten checkrun's CompletionTime
+				return cr.ID, false, nil
+			}
+		}
+	}
+	c.logger.Info("Found no CheckRuns with a matching ExternalID", "ExternalID", checkRun.ExternalID)
+
+	return nil, true, nil
+}
+
+func (c *Client) GetAllCommitStatusesForRef(ctx context.Context, owner, repo, sha string) ([]*ghapi.RepoStatus, error) {
+	res, _, err := c.GetRepositoriesService().ListStatuses(ctx, owner, repo, sha, &ghapi.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(res) == 0 {
+		c.logger.Info("Found no commitStatus for the ref", "SHA", sha)
+		return nil, nil
+	}
+
+	return res, nil
+}
+
+// GetCommitStatusID returns if a match is found for the SHA, externalID,appID, status, context and decription.
+func (c *Client) HasCommitStatusExist(res []*ghapi.RepoStatus, commitStatus *CommitStatusAdapter) (bool, error) {
```

Comment:
`HasCommitStatusExist` should be probably reworded, `Has` sounds extra there, maybe just `CommitStatusExists`

Reply:
fixed.